### PR TITLE
Backport EventEngine Forkables

### PR DESCRIPTION
**Pull request by drfloob**
This fixes fork-related process hangs due to threads not being properly shutdown.

### CODE REVIEW

File: src/core/lib/event_engine/forkable.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+#include "absl/container/flat_hash_set.h"
+
+#include "src/core/lib/gprpp/sync.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+namespace {
+grpc_core::Mutex g_mu;
```
Comment (ctiller) [inline src/core/lib/event_engine/forkable.cc:31]: NoDestruct<>

File: src/core/lib/event_engine/forkable.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+bool g_registered{false};
+absl::flat_hash_set<Forkable*> g_forkables;
```
Comment (ctiller) [inline src/core/lib/event_engine/forkable.cc:33]: NoDestruct<>

File: src/core/lib/event_engine/forkable.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+#include "absl/container/flat_hash_set.h"
+
+#include "src/core/lib/gprpp/sync.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+namespace {
+grpc_core::Mutex g_mu;
+bool g_registered{false};
```
Comment (ctiller) [inline src/core/lib/event_engine/forkable.cc:32]: ABSL_GUARDED_BY(..)

File: src/core/lib/event_engine/forkable.cc (hunk 2 above)
Comment (ctiller) [reply src/core/lib/event_engine/forkable.cc:33]: and ABSL_GUARDED_BY(..)

**Review by ctiller — APPROVED** [review]

File: src/core/lib/event_engine/forkable.cc (hunk 1 above)
Comment (drfloob) [reply src/core/lib/event_engine/forkable.cc:31]: NoDestruct didn't exist in v1.48. I'll backport it as well.

Comment (drfloob) [reply src/core/lib/event_engine/forkable.cc]: Done

File: src/core/lib/event_engine/forkable.cc (hunk 2 above)
Comment (drfloob) [reply src/core/lib/event_engine/forkable.cc]: Done

File: src/core/lib/event_engine/forkable.cc (hunk 3 above)
Comment (drfloob) [reply src/core/lib/event_engine/forkable.cc]: Done

**Review by ctiller — APPROVED** [review]

**Pull request merged**